File: 1544-make-the-string-great/1544-make-the-string-great.java
class Solution {
    public String makeGood(String s) {
        StringBuilder sb = new StringBuilder();
        Stack<Character> stack = new Stack<>();
        for(char c : s.toCharArray()){
            if(stack.isEmpty()){
                stack.push(c);
            }
            else{
                char peek = stack.peek();
                if(Character.isUpperCase(peek) && Character.isLowerCase(c)
                  || Character.isLowerCase(peek) && Character.isUpperCase(c)){
                    if(Character.toLowerCase(peek) == Character.toLowerCase(c)){
                        stack.pop();
                        continue;
                    }else{
                        stack.push(c);
                    }
                }else{
                    stack.push(c);
                }
            }
        }
        while(!stack.isEmpty()){
            sb.insert(0, stack.pop());
        }
        return sb.toString();
    }
}

File: 1544-make-the-string-great/1544-make-the-string-great.py
class Solution:
    def makeGood(self, s: str) -> str:
        stack = []
        for c in s: 
            if len(stack) == 0:
                stack.append(c)
            else:
                peek = stack[-1]
                if peek.isupper() and c.islower() or c.isupper() and peek.islower():
                    if peek.lower() == c.lower():
                        stack.pop()
                        continue
                    else:
                        stack.append(c)
                else:
                    stack.append(c)
        return ''.join(stack)